.circleci/shellcheck.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+#!/usr/bin/env bash
+# Sets some Bash options to encourage well formed code.
+# For example, some of the options here will cause the script to terminate as
+# soon as a command fails. Another option will cause an error if an undefined
+# variable is used.
+# See: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html
+
+# Any trap on ERR is inherited by shell functions, command substitutions, and
+# commands executed in a subshell environment. The ERR trap is normally not
+# inherited in such cases.
+set -o errtrace
```

Comment:
Why are all these copy/pasted into this script instead of using `bootstrap.sh`?

Reply:
It's a trade-off. On the one hand, you're right, it's a bit awkward to mix the code we're testing with the code we're using to test it. On the other hand, every time we update `bootstrap.sh`, we'll end up either (a) updating `shellcheck.sh` too, in which case, it seems pointless to keep it separate or (b) we'll forget to update `shellcheck.sh`, in which case, it may end up causing bugs in and of itself.

Reply:
Seems fair to me. I'll source `bootstrap.sh` from `shellcheck.sh`.